lib/hermann/producer.rb

```diff
@@ -62,7 +62,7 @@ def push(value, opts={})
       end
 
       if Hermann.jruby?
-        result = @internal.push_single(value, topic, nil)
+        result = @internal.push_single(value, topic, opts.fetch(:key, ""))
```

Comment:
Are you sure the behavior of passing an empty string in here is safe? I'm not sure what the machinery underneath `KeyedMessage` would do with an empty string.

I want to make sure that in the default case, no supplied `:key`, that the random partitioning is used underneath